### PR TITLE
Add the ability to run Starlark during debug dumps, and run debug dumps locally

### PR DESCRIPTION
**Pull request by jrockway**
This adds a (placeholder) ability to run Starlark during debug dumps, and a pachctl command to run the dump locally.  I also fixed a formatting bug with "pachctl debug template", which was using the wrong YAML encoder.

This is part 2 of the Starlark debug dumps project; part 1 was adding the Starlark infrastructure, and part 3 is actually adding useful functionality to Starlark debug dumps.

For developers developing scripts, use `starpach run -p debugdump` or `starpach shell -p debugdump` to develop them.  Then put the script in a template, give it to a user, and have them run `pachctl debug local template.yaml`.  To have a script always run, add it to `src/server/debug/server/debugstar/starlark` and it will be built into pachd and pachctl for use with the `builtin` option in the template.  (They are run by default on the server.)

### CODE REVIEW

File: src/debug/debug.proto [hunk 1]
```diff
@@ -88,29 +88,58 @@ message App {
...
   repeated App loki_logs = 6;
   repeated App binaries = 7;
   repeated App profiles = 8;
 }
 
+// StarlarkScript is a Starlark script.
+message StarlarkScript {
+  // The name of the script; used for debug messages and to control where the output goes.
+  string name = 1;
+  // The text of the "debugdump" peronality Starlark program.
```
Comment (robert-uhl) [inline src/debug/debug.proto:109]: s/peronality/personality

File: src/debug/debug.proto [hunk 2]
```diff
@@ -88,29 +88,58 @@ message App {
...
+  // How long to run the dump for.
   google.protobuf.Duration timeout = 4;
   message Defaults {
+    // If true, include the cluster defaults.
     bool cluster_defaults = 1;
   }
+  // Which defaults to include in the debug dump.
   Defaults defaults = 5;
+  // A list of Starlark scripts to run.
+  repeated Starlark starlark_scripts = 6;
```
Comment (robert-uhl) [inline src/debug/debug.proto:142]: The naming here is slightly confusing: `starlark_scripts` is a sequence of `Starlark` messages, which may refer to either builtins or `StarlarkScript` messages.

Maybe `starlark_scripts` is a sequence of `StarlarkScript` messages, which may refer to either builtins or a `StarlarkSource` message?

File: src/server/debug/server/debugstar/debugstar_test.go [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+	"testing/fstest"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pctx"
+)
+
+type testDumpFS fstest.MapFS
+
+var _ DumpFS = (*testDumpFS)(nil)
```
Comment (acohen4) [inline src/server/debug/server/debugstar/debugstar_test.go:21]: 👍

**Review by acohen4 — APPROVED** [review]
LGTM

File: src/debug/debug.proto (hunk 1 above)
Comment (jrockway) [reply src/debug/debug.proto:109]: Done.

File: src/debug/debug.proto (hunk 2 above)
Comment (jrockway) [reply src/debug/debug.proto:142]: I made the other thing a `StarlarkLiteral`, so you can have a builtin or literal script.  I think that makes sense.

**Pull request merged**